### PR TITLE
Code blocks

### PR DESCRIPTION
**Pull request by Abdih10**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -18,6 +18,7 @@ target/
 *.iws
 *.iml
 *.ipr
+*.DS_Store
```
Comment (DoNotWorr) [inline .gitignore:23]: This is not related to IntelliJ IDEA.

Comment (DoNotWorr) [reply .gitignore:23]:
```suggestion

### MacOS ###
*.DS_Store
```

File: src/main/java/CreateFiles/createTempFile.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+           Path temp = Files.createTempFile(path, prefix, ".log");
+
+           System.out.println("Temp file : " + temp);
+
+       } catch (IOException e) {
+           System.err.println("Sorry, there was a problem creating temporary file or directory");
+           e.printStackTrace();
+       }
+   }
+}
```
Comment (DoNotWorr) [inline src/main/java/CreateFiles/createTempFile.java]: I think you mentioned this class should be deleted.

File: src/main/java/org/five/sonarqubot/events/FileService.java [hunk 3]
```diff
@@ -2,6 +2,10 @@
 
 import reactor.core.publisher.Mono;
 
+
```
Comment (DoNotWorr) [inline src/main/java/org/five/sonarqubot/events/FileService.java:5]: Don't add empty lines.

I think you should run autoformat on the entire project to make sure there are no other weird formatting.

File: src/main/java/org/five/sonarqubot/events/FileServiceImpl.java [hunk 4]
```diff
@@ -1,12 +1,71 @@
...
+            }
+            System.out.println("File already exists");
+
+            writeToFile(code, file);
+
+        } catch (IOException e) {
+            System.err.println("Sorry, there was a problem creating temporary file");
+            e.printStackTrace();
+        }
+        return Mono.just(code);
```
Comment (DoNotWorr) [inline src/main/java/org/five/sonarqubot/events/FileServiceImpl.java]: It would be better if you return filename in the Mono (instead of code text).

File: src/main/java/org/five/sonarqubot/events/MessageServiceImpl.java [hunk 5]
```diff
@@ -10,6 +10,11 @@ public class MessageServiceImpl implements MessageService {
...
         return Mono.just(eventMessage)
                 .filter(message -> message.getAuthor().map(user -> !user.isBot()).orElse(false))
-                .filter(message -> message.getContent().equalsIgnoreCase("alex"));
+                .filter(message -> message.getContent().startsWith("```") && message.getContent().endsWith("```"));
     }
-}
+
+    @Override
+    public Mono<String> getCode(Message message) {
+        return Mono.just(message.getContent().split("```")[1].trim());
```
Comment (DoNotWorr) [inline src/main/java/org/five/sonarqubot/events/MessageServiceImpl.java:18]: This code assumes there is only one codeblock in the message.
If a user sends badly formatted messages that for example contains:

\`\`\`
hejhejhej
\`\`\`
/* valid code */
\`\`\`

...then our solution would analyze "hejhejhej".

Comment (DoNotWorr) [reply src/main/java/org/five/sonarqubot/events/MessageServiceImpl.java:18]: You decide if this should be fixed right now.
We could simply make a Trello task to fix it later.

**Review by DoNotWorr — CHANGES_REQUESTED** [review]
Some formatting stuff, comments etc

**Review by IsaLjungdahl — APPROVED** [review]

**Review by DoNotWorr — APPROVED** [review]
🎖️
